.github/workflows/docker.yml

```diff
@@ -0,0 +1,145 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - "master"
```

Comment:
We might want to add in `pull_request:` so we can test that a PR doesn't break any images, plus it would be good to see it running before we merge.

Reply:
We should of course only push the images on merge to master.

Reply:
If we do this, should we set up a separate repo or tag so that the PR doesn't overwrite master images?

Reply:
We should only do the build part ...

Reply:
Fixed by setting `push: ${{ github.event_name != 'pull_request' }}` under all of the `docker/build-push-action@v3` actions